vllm/entrypoints/openai/rpc/server.py

```diff
@@ -34,11 +30,31 @@ def __init__(self, async_engine_args: AsyncEngineArgs,
         # see https://stackoverflow.com/a/8958414
         self.socket.bind(f"tcp://127.0.0.1:{port}")
 
+        self.async_engine_args = async_engine_args
+        self.usage_context = usage_context
+
     def cleanup(self):
         """Cleanup all resources."""
         self.socket.close()
         self.context.destroy()
 
+    async def startup_engine(self, identity):
```

Comment:
I agree that the code of this approach is cleaner, but I'm worried about crashes outside of the python exception flow. Since the server loads C libraries it's not immune to segmentation faults, bus errors and other low level errors that can happen especially when we're dealing with hardware accelerators. I think it's more robust if we have a loop on the client side that monitors the server process.

Reply:
Sounds good. I’ll close this